File: DeJia/app/src/main/java/com/dejia/anju/api/UpLoadUgcImageApi.java
package com.dejia.anju.api;

import android.content.Context;

import com.dejia.anju.api.base.BaseCallBackListener;
import com.dejia.anju.api.base.BaseCallBackUploadApi;
import com.dejia.anju.net.NetWork;
import com.lzy.okgo.model.HttpParams;

import java.util.Map;

//上传文章图片接口
public class UpLoadUgcImageApi implements BaseCallBackUploadApi {
    @Override
    public void getCallBack(Context context, Map<String, Object> maps, HttpParams uploadPamas, final BaseCallBackListener listener) {
        NetWork.getInstance().call("ugc", "uploadImage", maps, uploadPamas, mData -> listener.onSuccess(mData));
    }
}

File: DeJia/app/src/main/java/com/dejia/anju/api/UpLoadUgcImageApi.kt
package com.dejia.anju.api

import android.content.Context
import com.dejia.anju.api.base.BaseCallBackListener
import com.dejia.anju.api.base.BaseCallBackUploadApi
import com.dejia.anju.net.NetWork.Companion.instance
import com.dejia.anju.net.ServerCallback
import com.dejia.anju.net.ServerData
import com.lzy.okgo.model.HttpParams

//上传文章图片接口
class UpLoadUgcImageApi : BaseCallBackUploadApi {
    override fun getCallBack(context: Context, maps: MutableMap<String, Any>, uploadPamas: HttpParams, listener: BaseCallBackListener<*>) {
        instance!!.call("ugc", "uploadImage", maps, uploadPamas, context, ServerCallback { mData: ServerData? -> listener.onSuccess(mData) })
    }
}